### DISCUSSION TITLE
Welcome to StableStudio Discussions!

### DISCUSSION THREAD

**Discussion by cruhl** [Stability-AI/StableStudio#1] [Announcements]
Hello, World!

**Comment by kopyl** [comment]
What framework do you use for it? Next? Angular? React? Vue? Why there is no info on it?

**Reply by GastonDonnet** [reply]
Check the code: https://github.com/Stability-AI/StableStudio/tree/main/packages/stablestudio-ui

[TypeScript](https://www.typescriptlang.org/) – Type safety ftw!

[Vite](https://vitejs.dev/) – Bundler and live-development tool. You can see the full config in [vite.config.ts](https://github.com/Stability-AI/StableStudio/blob/main/packages/stablestudio-ui/vite.config.ts)

[React](https://reactjs.org/) –  We're using React with modern hooks and functional components

[Zustand](https://docs.pmnd.rs/zustand/getting-started/introduction) –  Extremely fast and easy-to-use state management, it's powerful and tiny!

[Tailwind](https://tailwindcss.com/) – Our prefered method of styling components, the config is at [tailwind.config.js](https://github.com/Stability-AI/StableStudio/blob/main/packages/stablestudio-ui/tailwind.config.cjs)

[Emotion](https://emotion.sh/docs/introduction) – For when we need to break free of Tailwind and write "native CSS"

**Reply by kopyl** [reply]
Thanks :)

**Comment by mhsabbagh** [comment]
Hello.

Is my understanding correct that the current version of StableStudio only supports using the DreamStudio API for image generation (i.e. running the models), and does not support using 3rd-party/local instances of the models (but it's something on your roadmap? Correct?

**Reply by cruhl** [reply]
That's correct, someone might beat us to it though! ~> https://github.com/Stability-AI/StableStudio/issues/25

**Comment by GeekyGhost** [comment]
Hello from the world of tomorrow! lol

**Comment by Zengrath** [comment]
What is the hardware requirements? Nvidia or CPU only? Or does it work with AMD too? If AMD is it only some RDNA2 cards or is the 7900 xtx  RDNA3 supported?

**Reply by cruhl** [reply]
Since this is just a UI almost any computer should be able to run it, that might change with local inference. ~> https://github.com/Stability-AI/StableStudio/issues/25

**Comment by RavianXReaver** [comment]
What's the actual purpose of this project if we're still using dreamstudio to generate images? I thought we were able to use readily available models. Maybe I misunderstood the whole purpose of this project. Please correct me if i'm wrong.

**Reply by GeekyGhost** [reply]
That's part of what they wanted help working out. They've released it under an MIT license so these types of things can be added.

**Reply by cruhl** [reply]
Going forward, when we add new things to DreamStudio, we'll be developing them in the open here.

The idea is making this project as cool as possible with direct Stability contributions and community contributions.

**Comment by RavianXReaver** [comment]
Understood! thanks :)

**Comment by sancoderhost** [comment]
Does it supports custom models?

**Reply by KAJdev** [reply]
StableStudio is not opinionated on the backend.

**Comment by AmmarAhmed1735** [comment]
what are the plans for community contributions in the future?

**Reply by GeekyGhost** [reply]
I have personally hit a snag in personal life, so I'm working on what I can lol. First is basically offline I think and then integrating features. I want to work in Riffusion if it isn't planned and etc. for example.

**Comment by AmmarAhmed1735** [comment]
I see, that is really interesting. Could you tell me more about Riffusion?
I am also looking to gain some experience in it.

**Reply by GeekyGhost** [reply]
Oops, sorry. I am all over the place lol. https://github.com/enlyth/sd-webui-riffusion   It has a bit of an issue with automatics torch version, if you go to the issues page though, I rewrote the install.py. If you manually clone it to your extension folder, then edit the install.py file, it will launch and load and everything fine. Seems to work well with the current torchaudio version with automatic1111. Haven't tried it with TRT, but if you use Multidiffusion you can get Spectrograms 8192 wide and then take them to audacity and slow it down to like 89, change the bass and treble levels and it sounds like a 512 spectrogram. If you want to bring out more of the sound add some reverb. Then you can hear more of the instruments and etc in the Spectrogram. Audacity is free and even has a Spectrogram viewer built in for the Audio. It's interesting to see the differences and etc.
